src/causalprog/graph/graph.py

```diff
@@ -50,10 +50,12 @@ def add_edge(self, first_node: Node | str, second_node: Node | str) -> None:
             self.add_node(first_node)
         if second_node.label not in self._nodes_by_label:
             self.add_node(second_node)
-        for node_to_check in (first_node, second_node):
-            if node_to_check != self._nodes_by_label[node_to_check.label]:
-                msg = "Invalid node: {node_to_check}"
-                raise ValueError(msg)
+        if first_node != self._nodes_by_label[first_node.label]:
+            msg = "Invalid node"
+            raise ValueError(msg)
+        if second_node != self._nodes_by_label[second_node.label]:
+            msg = "Invalid node"
+            raise ValueError(msg)
```

Comment:
```suggestion
        for node_to_check in (first_node, second_node):
            if node_to_check != self._nodes_by_label[node_to_check.label]:
                msg = "Invalid node: {node_to_check}"
                raise ValueError(msg)
```

Looks like you're loosing some of the changes that were made to #16 in the review comments, before we started this during pair-programming. This shouldn't be changing back.